### PR TITLE
Fix components enabled config option when merging with defaults.

### PR DESCRIPTION
**Pull request by dereuromark**
Resolves https://github.com/cakephp/cakephp/issues/9660 for 3.x

...Checking if tests pass.

### CODE REVIEW

**Comment by markstory** [comment]
Tests?

**Comment by dereuromark** [comment]
If someone has time feel free to add some on top. Otherwise next week I will have more time at the latest.

**Comment by thinkingmedia** [comment]
@dereuromark I think that's all.

**Review by markstory — APPROVED** [review]
Looks good to me.

File: tests/test_app/TestApp/Controller/Component/SomethingWithCookieComponent.php [hunk 1]
```diff
@@ -17,6 +17,8 @@
 
 /**
  * SomethingWithCookieComponent class
+ *
+ * @property Component\CookieComponent $Cookie
```
Comment (dereuromark) [inline tests/test_app/TestApp/Controller/Component/SomethingWithCookieComponent.php]: I think that is incorrect from the namespace
also should be FQCN but no big deal here

Comment (thinkingmedia) [inline tests/test_app/TestApp/Controller/Component/SomethingWithCookieComponent.php]: @dereuromark yeah, that doesn't look right. I'll fix it.

**Pull request merged**